dask/array/utils.py

```diff
@@ -128,8 +130,10 @@ def compute_meta(func, _dtype, *args, **kwargs):
             ):
                 raise
             else:
+                np.seterr(**np_err)
                 return None
         except Exception:
+            np.seterr(**np_err)
             return None
```

Comment:
You might consider using the `finally:` block here

```python
try:
    ...
except:
    ...
finally:
    np.seterr(**np_err)
```

The finally block is run regardless of whether or not there was an error.

Reply:
Sure, but there's more code after this block and we still want to keep warnings disabled until just before the last return statement.

Reply:
Instead of handling this ourselves, we could also just use:

```
with np.errstate(all="ignore"):
    ...
```

which handles setting and resetting for you.

Reply:
+1

Reply:
Thanks for the suggestion @jcrist, it's indeed a much cleaner solution!

Fixed in latest commits.